pgcli/pgclirc

```diff
@@ -187,6 +188,11 @@ output.null = "#808080"
 [alias_dsn]
 # example_dsn = postgresql://[user[:password]@][netloc][:port][/dbname]
 
+# Specify sperate history file for alias_dsn.
+# If not set, will use history_file settings.
+[dsn_history_file]
```

Comment:
We can support mapping of history file to not only DSN aliases, but database names as well. That way, if you have `mydb1` hosted on different servers, you could just configure:

```
[db_history_file]
mydb1 = ~/config/pgcli/history_mydb1
```

this could be useful to users that do not use DSN aliases.

There will be ambiguity if we do that: we'd have to figure out if `mydb1` is a DSN alias or a database name. We can look for matching DSN alias first, if user provided `--dsn` when running the cli, and if there's no alias found, assume this is database name.

Reply:
Yes, I like the idea that mapping of the history file for the database. I think it covers most of the use cases. Maybe consider `alias_dsn` is kind of overthinking and over-designed, I think we add too mush to configuration, and cloud leads to misunderstanding. I don't think users want to share history between databases. Even if they want to do that, they can use `--histfile` option.

Let's see, we only add a flag `sperate_history_on_database_name = False`, and add `--histfile` option. (Also, if user use `\c[onnect] database_name`, we should switch history file as well.

If the user does want to share history between database, they can:

1. Use `--histfile`, maybe use shell's alias together.
2. Save the most used query to `named query`.

What do you think?

Reply:
I think that:

* separating history per database does not need a config flag. Once that feature is in, we should do it by default. When you're connected to a specific database, queries from other databases are not very useful / relevant, and would only clutter history.
* user can use `--histfile` to override the default history-file-per-database.
* mapping to DSN vs database name can be useful. I had a use case with `prod` and `beta` database tables named `table1_prod`, `table2_prod` etc. Table structure was the same, but table names were different, so you could not use exactly the same query on those databases. This may be an edge case.

Reply:
I am convinced, I will only add a section named `dsn_history_file` section(maybe we cloud come up with a better name), and left other configs untouched.

So the process of choosing a history file will look like this:

![image](https://user-images.githubusercontent.com/9675939/96358270-e8b54780-1137-11eb-849f-8e345e8a4244.png)


Do I understand it correctly?

Reply:
Yes, this looks exactly right.